arch/arm/mach-zynqmp/cmd_bootslot.c

```diff
@@ -0,0 +1,259 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright (C) 2022 Foundries.io Ltd
+ */
+
+#include <common.h>
+#include <command.h>
+#include <dm.h>
+#include <dm/device-internal.h>
+#include <asm/io.h>
+#include <spi.h>
+#include <spi_flash.h>
+
+#define SYS_CHECKSUM_OFFSET		(0x3U)
+#define PERSISTENT_REG_SIZE		(0x20000U)
+#define ENV_VAR_PREFIX			"zynqmp."
+
+/* The below enums denote persistent registers in Qspi Flash */
+struct sys_persistent_state {
```

Comment:
all members of this struct are limited to {0,1}. Can we use a packed struct or even just a regular uint var to store all these variables as storage of flags?

Reply:
We can't change this struct,as `imgsel` app expects this format of data to be flashed on QSPI (this is the first bootloader preflashed to 0x0 offset on QSPI on Kria K260 Starter Kit) https://github.com/Xilinx/embeddedsw/tree/master/lib/sw_apps/imgsel

I've just re-implemented in U-Boot what is done for Linux userspace https://github.com/Xilinx/linux-image_update/blob/master/image_update.c

Reply:
Ah, sorry, I missed your description :)